app/src/main/java/org/fossasia/phimpme/opencamera/Camera/CameraActivity.java

```diff
@@ -2855,6 +2860,13 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String permissi
                         Log.d(TAG, "camera permission granted");
                     preview.retryOpenCamera();
                 } else {
+                    if (ActivityCompat.shouldShowRequestPermissionRationale(CameraActivity.this,
+                            Manifest.permission.CAMERA)) {
+                        // now, user has denied permission (but not permanently!)
+
+                    } else {
+                        permanentDenyPermission = true;
+                    }
```

Comment:
There is no need to add braces here, it is a single statement. Just type it in front of `else`.

Reply:
i got another idea , will push in a minute.